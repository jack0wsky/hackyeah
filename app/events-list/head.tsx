export default async function Head() {
    return <title>Events</title>
}