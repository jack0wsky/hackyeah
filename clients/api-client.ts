import axios from "axios";
import { QueryClient } from "react-query";

export const BASE_URL = "https://zygmuntd.eu.pythonanywhere.com";

export const client = new QueryClient();

export const apiClient = axios.create({
  baseURL: BASE_URL,
});

export class ApiClient {
  private readonly baseUrl: string;

  constructor() {
    this.baseUrl = BASE_URL;
  }

  getEvents(): string {
    return `${this.baseUrl}/events/`;
  }

  async createEvent() {
    await apiClient.post("/user/event/", {
      city: "",
      name: "",
      address: "",
      date_from: "",
      date_to: "",
      products: [],
    });
  }
}
