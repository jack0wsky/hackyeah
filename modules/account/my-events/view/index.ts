export * from "./EventsList";
export * from "./EventsListUI";
