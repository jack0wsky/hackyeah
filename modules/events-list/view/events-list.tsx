import Layout from "../../shared/layout";
import Button from "../../shared/button";
import Filters from "../components/filters";
import List from "../components/list";

export const EventsList = () => {
  return (
    <Layout title="Events">
      <main className="screen-size flex flex-col">
        <div className="w-full min-h-[108px] flex flex-col items-center md:flex-row justify-between p-24 bg-primary-blue/10 rounded-8 mt-24">
          <div>
            <p className="text-dark-blue uppercase text-[12px]">
              our less waste initiative
            </p>
            <p className="text-dark-blue text-[32px] font-light">
              Leave what you <span className="font-bold">do not want</span> and
              take what <span className="font-bold">you need</span>
            </p>
          </div>

          <Button variant="secondary">Find out more</Button>
        </div>

        <Filters />

        <List />
      </main>
    </Layout>
  );
};
