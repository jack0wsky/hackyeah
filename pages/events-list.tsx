import Layout from "../components/shared/layout";
import Filters from "../components/events-list/filters";
import List from "../components/events-list/list";

const EventsList = () => {
  return (
    <Layout>
      <main className="screen-size flex flex-col">
        <Filters />

        <List />
      </main>
    </Layout>
  );
};

export default EventsList;
