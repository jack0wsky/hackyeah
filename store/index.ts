import { configureStore } from "@reduxjs/toolkit";
import { useSelector, useDispatch } from "react-redux";
import {
  updateSearch,
  updateStartDate,
  updateEndDate,
  toggleTypes,
  updateStartTime,
  updateEndTime,
  updateCity,
} from "./slices/filters-slice";
import filtersReducer from "./slices/filters-slice";
import type { ItemType } from "../types";

export const store = configureStore({
  reducer: {
    filters: filtersReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;

export const useStore = () => {
  const state = useSelector((state: RootState) => state);
  const dispatch = useDispatch();

  return {
    ...state,
    updateCity: (city: string) => dispatch(updateCity(city)),
    updateSearch: (phrase: string) => dispatch(updateSearch(phrase)),
    updateStartTime: (time: string) => dispatch(updateStartTime(time)),
    updateEndTime: (time: string) => dispatch(updateEndTime(time)),
    toggleTypes: (option: ItemType) => dispatch(toggleTypes(option)),
    updateStartDate: (startDate: string) =>
      dispatch(updateStartDate(startDate)),
    updateEndDate: (endDate: string) => dispatch(updateEndDate(endDate)),
  };
};
