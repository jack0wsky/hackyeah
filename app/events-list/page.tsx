'use client'

import { EventsList } from "../../modules/events-list/view/events-list";

export default EventsList;
