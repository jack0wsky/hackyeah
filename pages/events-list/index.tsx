import React from "react";

import { EventsList } from "@/modules/events-list/view/events-list";

export default EventsList;
