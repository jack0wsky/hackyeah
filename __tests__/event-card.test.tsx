import React from "react";
import "@testing-library/jest-dom";
import { EventsList } from "../modules/events-list/view/events-list";
import Filters from "../modules/events-list/components/filters";
import { Provider } from "react-redux";
import { QueryClientProvider } from "react-query";
import { client } from "../clients/api-client";
import { store } from "../store";
import { getByPlaceholderText, render } from "@testing-library/react";
import { toJSON } from "yaml/util";

describe(EventsList.name, () => {
  const component = render(
    <Provider store={store}>
      <QueryClientProvider client={client}>
        <Filters />
      </QueryClientProvider>
    </Provider>
  );

  it("renders list of events", async () => {
    const button = await getByPlaceholderText(
      component.baseElement,
      "Search..."
    );

    button

    expect(toJSON(component.baseElement)).toMatchInlineSnapshot(`""`);
  });
});
