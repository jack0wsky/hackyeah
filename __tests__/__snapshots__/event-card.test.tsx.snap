// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsList renders list of events: "" 1`] = `
<p
  data-testid="test"
>
  something
</p>
`;
